x/staking/types/validator.go

```diff
@@ -384,6 +374,7 @@ func (v Validator) RemoveDelShares(pool Pool, delShares sdk.Dec) (Validator, Poo
 	}
 
 	v.DelegatorShares = remainingShares
+	// TODO: This is unobvious from the name of the function, either justify or move outside.
```

Comment:
ditto

Reply:
Ditto, it depends on whether the tokens are *going*, e.g. in `keeper.Slash` we have to update the pool again afterwards to burn them

Reply:
I sustain - that seems reasonable to burn them afterwards and leave as is here. This function simply removes the tokens from the validator and further allows the caller function to determine their further destination... this logical flow is nice and consolidates common functionality.